src/test/java/edu/ucla/library/prl/harvester/JobRequestsFT.java

```diff
@@ -0,0 +1,128 @@
+
+/**
+ * Tests the application's behavior in response to requests involving {@link Job}s.
+ */
+@ExtendWith(VertxExtension.class)
+@TestInstance(Lifecycle.PER_CLASS)
+public class JobRequestsFT {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JobRequestsFT.class, MessageCodes.BUNDLE);
+
+    private static final UriTemplate INSTITUTIONS = UriTemplate.of("/institutions");
+
+    private static final UriTemplate JOBS = UriTemplate.of("/jobs");
+
+    private Pool myDbConnectionPool;
+
+    private WebClient myWebClient;
+
+    private int myInstitutionID;
+
+    /**
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @BeforeAll
+    public final void setUp(final Vertx aVertx, final VertxTestContext aContext) {
+        ConfigRetriever.create(aVertx).getConfig().compose(config -> {
+            final String host = config.getString(Config.HTTP_HOST);
+            final int port = config.getInteger(Config.HTTP_PORT);
+
+            myDbConnectionPool = HarvestScheduleStoreService.getConnectionPool(aVertx, config);
+            myWebClient = WebClient.create(aVertx, new WebClientOptions().setDefaultHost(host).setDefaultPort(port));
+
+            return Future.succeededFuture();
+        }).onSuccess(result -> aContext.completeNow()).onFailure(aContext::failNow);
+    }
+
+    /**
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     * @throws AddressException
+     * @throws MalformedURLException
+     * @throws NumberParseException
+     */
+    @BeforeEach
+    public void beforeEach(final Vertx aVertx, final VertxTestContext aContext)
+            throws AddressException, MalformedURLException, NumberParseException {
+        myWebClient.post(INSTITUTIONS).sendJson(TestUtils.getRandomInstitution().toJson()).onSuccess(response -> {
+            myInstitutionID = new Institution(response.bodyAsJsonObject()).getID().get();
+
+            aContext.completeNow();
+        }).onFailure(aContext::failNow);
+    }
+
+    /**
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @AfterEach
+    public void afterEach(final Vertx aVertx, final VertxTestContext aContext) {
+        TestUtils.wipeDatabase(myDbConnectionPool).onSuccess(nil -> aContext.completeNow())
+                .onFailure(aContext::failNow);
+    }
+
+    /**
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @AfterAll
+    public final void tearDown(final Vertx aVertx, final VertxTestContext aContext) {
+        myWebClient.close();
+        myDbConnectionPool.close().onSuccess(result -> aContext.completeNow()).onFailure(aContext::failNow);
+    }
+
+    /**
+     * Tests that {@link Op#listJobs} initially retrieves an empty list.
+     *
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @Test
+    void testListBeforeAdd(final Vertx aVertx, final VertxTestContext aContext) {
```

Comment:
Uh... don't you need a test to prove the handler actually retrieves jobs when the db is populated?

Reply:
This is why I implemented all the Institution ops in #35; because I think verifying each method's behavior is most clearly done in context with all the other methods. It seems like a chicken-and-egg problem: it's a lot easier to see that listJobs, getJob, updateJob, and removeJob are doing the right thing if you can addJob. But since [the last PR was too big](https://github.com/UCLALibrary/prl-harvester/pull/35#issuecomment-1403831391), I figured it would be better to scope each PR to a single ticket. However I can add more stuff if this PR is too bare.

Reply:
@ksclarke What do you think? Do you want these changesets scoped to each Jira ticket, or OK to implement addJob in this PR?

Reply:
Up to you. In reflecting on that last ticket, I may have been overreacting on the size of it. It was nowhere near the size of some of the large PRs I've submitted (which should have been broken up). I think my reaction was in part due to the lumping approach of the tests, which seemed like a paradigm shift rather than a large amount of code, really.

Reply:
Ok, I'll go ahead and add it in then.

Reply:
(It was mostly stashed, anyway.)